From 1dc4778baeefb0459edd51a56ea4050607bdb7c5 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:45 -0400
Subject: [powerpc] nvram: Scan partitions only once

Message-id: <1410545655-205645-177-git-send-email-dzickus@redhat.com>
Patchwork-id: 94276
O-Subject: [RHEL7 PATCH 176/626] powerpc/nvram: Scan partitions only once
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 65f36f4149ce4490b2174ef45e1dbe494f4b8268
Author: Cedric Le Goater <clg@fr.ibm.com>
Date:   Wed Oct 30 14:47:06 2013 +0100

    powerpc/nvram: Scan partitions only once

    nvram_scan_partitions() is called twice when initializing the "lnx,oops-log"
    partition and the "ibm,rtas-log" partition. This fills the partition list
    with duplicate entries. This patch moves the partition scan in the init
    routine pseries_nvram_init_log_partitions() which is called only once.

    Signed-off-by: Cedric Le Goater <clg@fr.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/nvram.c b/arch/powerpc/platforms/pseries/nvram.c
index ed14b05..2f91a9d 100644
--- a/arch/powerpc/platforms/pseries/nvram.c
+++ b/arch/powerpc/platforms/pseries/nvram.c
@@ -429,9 +429,6 @@ static int __init pseries_nvram_init_os_partition(struct nvram_os_partition
  loff_t p;
  int size;
 
- /* Scan nvram for partitions */
- nvram_scan_partitions();
-
  /* Look for ours */
  p = nvram_find_partition(part->name, NVRAM_SIG_OS, &size);
 
@@ -795,6 +792,9 @@ static int __init pseries_nvram_init_log_partitions(void)
 {
  int rc;
 
+ /* Scan nvram for partitions */
+ nvram_scan_partitions();
+
  rc = pseries_nvram_init_os_partition(&rtas_log_partition);
  nvram_init_oops_partition(rc == 0);
  return 0;
-- 
1.7.1